walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -24,64 +24,121 @@
+                if (current.contains(timeRange)) {
+                    continue;
+                }
+
+                if (current.overlaps(timeRange)) {
+                    current = timeRange;
+                    continue;
+                }
+
+                toAdd = TimeRange.fromStartEnd(current.end(), timeRange.start(), false);
+                if (toAdd.duration() >= request.getDuration()) {
+                    timeRanges.add(toAdd);
+                }
+                current = timeRange;
             }
 
-            if(current.overlaps(attendeeRelevantTimeRanges.get(i))){
-                current = attendeeRelevantTimeRanges.get(i);
-                continue;                
-            }    
+            // Add a TimeRange with a range from the end of the latest event to the end of the day if duration is valid
+            toAdd = TimeRange.fromStartEnd(current.end(), TimeRange.END_OF_DAY, true);
+            if (toAdd.duration() >= request.getDuration()) {
+                timeRanges.add(toAdd);
+            }
 
-            toAdd = TimeRange.fromStartEnd(current.end(), attendeeRelevantTimeRanges.get(i).start(), false);
-            if(toAdd.duration() >= request.getDuration()){
+        } else {
+            current = attendeeRelevantTimeRanges.get(0);
+
+            toAdd = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, current.start(), false);
+            if (toAdd.duration() >= request.getDuration()) {
                 timeRanges.add(toAdd);
             }
-            current = attendeeRelevantTimeRanges.get(i);
-        }
+            
+            for (TimeRange timeRange: attendeeRelevantTimeRanges) {
+                if (current.contains(timeRange)) {
+                    continue;
+                }
 
-        // Add a TimeRange with a range from the end of the latest event to the end of the day if duration is valid
-        toAdd = TimeRange.fromStartEnd(current.end(), TimeRange.END_OF_DAY, true);
-        if(toAdd.duration() >= request.getDuration()){
-            timeRanges.add(toAdd);
-        }
+                if (current.overlaps(timeRange)) {
+                    current = timeRange;
+                    continue;
+                }
+
+                toAdd = TimeRange.fromStartEnd(current.end(), timeRange.start(), false);
+                if (toAdd.duration() >= request.getDuration()) {
+                    timeRanges.add(toAdd);
+                }
+                current = timeRange;
+            }
 
+            toAdd = TimeRange.fromStartEnd(current.end(), TimeRange.END_OF_DAY, true);
+            if (toAdd.duration() >= request.getDuration()) {
+                timeRanges.add(toAdd);
+            }
+
+            List<TimeRange> timeRangesWithNoOptionalInterference = new ArrayList<TimeRange>();
+
+            // Goes through current timeRanges and removes meetings where optionalAttendees can't attend
+            for (int i = timeRanges.size() - 1; i >= 0; i--) {
```

Comment:
This algorithm will not work in all cases. Time ranges are removed if there is any overlap whatsoever.

Add and pass a test:
```
Events:            |-A-|            |-B-|
Optional:                          |-C--|
Options:  |---1---|    |-----2----|
With A and B required, but -C optional. I believe this algorithm as-is will produce
Options:  |---1---|    
Because it will disqualify 2 even if there is plenty of time and only a minor overlap at the end.